dtfs-central-api/src/v1/controllers/portal/gef-deal/update-deal.controller.js

```diff
@@ -3,48 +3,51 @@ const { findOneDeal } = require('./get-gef-deal.controller');
 const db = require('../../../../drivers/db-client');
 
 const updateDeal = async (dealId, update) => {
-  if (ObjectId.isValid(dealId)) {
+  try {
+    if (!ObjectId.isValid(dealId)) {
+      return { status: 400, message: 'Invalid Deal Id' };
+    }
+
     const collection = await db.getCollection('deals');
     const originalDeal = await findOneDeal(dealId);
-
-    console.info('Updating Portal GEF deal.');
-
     const dealUpdate = {
       ...originalDeal,
       ...update,
       updatedAt: Date.now(),
     };
-
     const findAndUpdateResponse = await collection.findOneAndUpdate(
       { _id: { $eq: ObjectId(String(dealId)) } },
       { $set: dealUpdate },
       { returnNewDocument: true, returnDocument: 'after' }
     );
 
-    console.info('Updated Portal GEF deal');
-
     return findAndUpdateResponse.value;
+  } catch (error) {
```

Comment:
Same as above also `generateMINActivities` `try-catch` will catch the exception.

Reply:
`generateMINActivities` `try-catch` won't catch it anymore because we're catching it here. Instead the `PUT /portal/gef/deals/activity/:id` endpoint will return a response where:
- the body is `{ status: 400, message: <the error> }`, but
- the status is `200` (even though the endpoint had an error),

which is a bug